pkg/controllers/workapplier/utils.go

```diff
@@ -0,0 +1,872 @@
+	return manifestObjHash == inMemberClusterObjLastAppliedManifestObjHash, nil
+}
+
+// shouldPerformPostApplyDriftDetection checks if post-apply drift detection should be performed.
+func shouldPerformPostApplyDriftDetection(applyStrategy *fleetv1beta1.ApplyStrategy) bool {
+	// Post-apply drift detection is performed if (and only if):
+	// * The apply strategy dictates that drift detection should run in full comparison mode.
+	return applyStrategy.ComparisonOption == fleetv1beta1.ComparisonOptionTypeFullComparison
+}
+
+// isManifestObjectApplied returns if an applied result type indicates that a manifest
+// object in a bundle has been successfully applied.
+func isManifestObjectApplied(appliedResTyp manifestProcessingAppliedResultType) bool {
+	return appliedResTyp == ManifestProcessingApplyResultTypeApplied ||
+		appliedResTyp == ManifestProcessingApplyResultTypeAppliedWithFailedDriftDetection ||
+		appliedResTyp == ManifestProcessingApplyResultTypeNoDiffFound
+}
+
+// isManifestObjectAvailable returns if an availability result type indicates that a manifest
+// object in a bundle is available.
+func isAppliedObjectAvailable(availabilityResTyp ManifestProcessingAvailabilityResultType) bool {
+	return availabilityResTyp == ManifestProcessingAvailabilityResultTypeAvailable || availabilityResTyp == ManifestProcessingAvailabilityResultTypeNotTrackable
+}
+
+// setManifestAppliedCondition sets the applied condition for a specific manifest.
+func setManifestAppliedCondition(
+	manifestCond *fleetv1beta1.ManifestCondition,
+	appliedResTyp manifestProcessingAppliedResultType,
+	applyError error,
+	inMemberClusterObjGeneration int64,
+) {
+	var appliedCond *metav1.Condition
+	switch {
+	case appliedResTyp == ManifestProcessingApplyResultTypeApplied:
+		// The manifest has been successfully applied.
+		appliedCond = &metav1.Condition{
+			Type:               fleetv1beta1.WorkConditionTypeApplied,
+			Status:             metav1.ConditionTrue,
+			Reason:             string(ManifestProcessingApplyResultTypeApplied),
+			Message:            ManifestProcessingApplyResultTypeAppliedDescription,
+			ObservedGeneration: inMemberClusterObjGeneration,
+		}
+	case appliedResTyp == ManifestProcessingApplyResultTypeAppliedWithFailedDriftDetection:
+		// The manifest has been successfully applied, but drift detection has failed.
+		//
+		// At this moment Fleet does not prepare a dedicated condition for drift detection
+		// outcomes.
+		appliedCond = &metav1.Condition{
+			Type:               fleetv1beta1.WorkConditionTypeApplied,
+			Status:             metav1.ConditionTrue,
+			Reason:             string(ManifestProcessingApplyResultTypeAppliedWithFailedDriftDetection),
+			Message:            ManifestProcessingApplyResultTypeAppliedWithFailedDriftDetectionDescription,
+			ObservedGeneration: inMemberClusterObjGeneration,
+		}
+	case appliedResTyp == ManifestProcessingApplyResultTypeNoDiffFound:
```

Comment:
this does not seem to be right when the apply Type is ApplyStrategyTypeReportDiff.  No diff does not mean applied == true